distribution/tools/keystore-cli/src/test/java/org/opensearch/common/settings/KeyStoreWrapperTests.java

```diff
@@ -328,13 +330,14 @@ private void possiblyAlterEncryptedBytes(
         byte[] encryptedBytes,
         int truncEncryptedDataLength
     ) throws Exception {
-        indexOutput.writeInt(4 + salt.length + 4 + iv.length + 4 + encryptedBytes.length);
-        indexOutput.writeInt(salt.length);
-        indexOutput.writeBytes(salt, salt.length);
-        indexOutput.writeInt(iv.length);
-        indexOutput.writeBytes(iv, iv.length);
-        indexOutput.writeInt(encryptedBytes.length - truncEncryptedDataLength);
-        indexOutput.writeBytes(encryptedBytes, encryptedBytes.length);
+        DataOutput io = EndiannessReverserUtil.wrapDataOutput(indexOutput);
```

Comment:
@nknize I am concerned that using `EndiannessReverserUtil.wrapDataXxx` would (potentially) generate temporary `EndiannessReverserDataOutput` /  `EndiannessReverserDataInput` wrappers ("garbage"), taking into account how often they are called, it may add up unnecessary pressure to GC. Wdyt?

Reply:
Yeah, this isn't the most ideal but it's only used in the KeyStoreWrapperTests and not actually in any critical path.

Reply:
For tests - totally agree, but Checkpoint also use wrapping, https://github.com/opensearch-project/OpenSearch/pull/1109/files#diff-f916f3d153eb8dada3c8be868ae55d050b1f24ce02343cea642f12bb0b0b8635R158, anyway probably not much we can do about that :(

Reply:
Ah, yes. (been working this PR for a while). Not much we can do about that since the Endianness change. Fortunately that's for BWC so should only be used during rolling upgrades then the new version will take over.

Reply:
:+1 thanks!